### ISSUE TITLE
[BUG] ObjectIdentifier does not return correct value for native

### ISSUE THREAD

**Issue by aalba6675** [wbond/asn1crypto#80]
Documentation: "ObjectIdentifier instances are converted to a unicode string in a dotted-integer format when .native is accessed."

Example:
```
# Will print: "value_name"
print(MyType('1.8.2.1.23').native)
```
Expecting dotted-integer so the answer should be "1.8.2.1.23", no?

**Comment by joernheissler** [comment]
```python
>>> curve = asn1crypto.keys.NamedCurve('secp256r1')
>>> curve.native
'secp256r1'
>>> curve.dotted
'1.2.840.10045.3.1.7'
```

**Comment by aalba6675** [comment]
Yes - that's weird
```
>>> from asn1crypto.core import ObjectIdentifier
>>> 
>>> class MyType(ObjectIdentifier):
...     _map = {
...         '1.8.2.1.23': 'value_name',
...         '1.8.2.1.24': 'other_value',
...     }
... 
>>> print(MyType('1.8.2.1.23').native)
1.8.2.1.23
>>> print(MyType('1.8.2.1.23').dotted)
1.8.2.1.23
```

**Comment by joernheissler** [comment]
I guess there's actually a bug here.
`native` should print the mapped name from what I understand, but it doesn't always.
More verbose example:

```python
>>> from asn1crypto.core import ObjectIdentifier
>>> 
>>> class MyType(ObjectIdentifier):
...     _map = {
...         '1.8.2.1.23': 'value_name',
...         '1.8.2.1.24': 'other_value',
...     }
... 
>>> foo = MyType('1.8.2.1.23')
>>> print(foo.native)
1.8.2.1.23
>>> print(foo.dotted)
1.8.2.1.23
>>> 
>>> foo2 = MyType.load(foo.dump())
>>> print(foo2.native)
value_name
>>> print(foo2.dotted)
1.8.2.1.23
>>> 
>>> bar = MyType('value_name')
>>> print(bar.native)
value_name
>>> print(bar.dotted)
1.8.2.1.23
>>> 
>>> bar2 = MyType.load(bar.dump())
>>> print(bar2.native)
value_name
>>> print(bar2.dotted)
1.8.2.1.23
```

[foo.py.txt](https://github.com/wbond/asn1crypto/files/1587282/foo.py.txt)

**Comment by aalba6675** [comment]
I think the bug is in the indentation level of native() method in ObjectIdentifier
```
    def native(self):
        """
        The a native Python datatype representation of this value

        :return:
            A unicode string or None. If _map is not defined, the unicode string
            is a string of dotted integers. If _map is defined and the dotted
            string is present in the _map, the mapped value is returned.
        """

        if self.contents is None:
            return None

        if self._native is None:
            self._native = self.dotted
            if self._map is not None and self._native in self._map:
                self._native = self._map[self._native]
        return self._native
```

The indentation looks wrong: it works if you outdent the last few lines:
```        if self._native is None:
            self._native = self.dotted
        if self._map is not None and self._native in self._map:
            self._native = self._map[self._native]
        return self._native
```

**Comment by aalba6675** [comment]
Test with outdent
```
[GCC 7.2.1 20170915 (Red Hat 7.2.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from asn1crypto import pkcs12
>>> y = pkcs12.CertId('x509')
>>> y.native
'x509'
>>> y.dotted
'1.2.840.113549.1.9.22.1'
>>> y = pkcs12.CertId('1.2.840.113549.1.9.22.1')
>>> y.native
'x509'
>>> y.dotted
'1.2.840.113549.1.9.22.1'
>>> 
```

**Comment by aalba6675** [comment]
PR https://github.com/wbond/asn1crypto/pull/81

**Comment by wbond** [comment]
This should be resolved by your fix, merged in b3e7eed8944fc61471ec9d2900ec679daef4c5f2
